Adapt VA-HCI for shiftstack

rlobillo opened the pull request:
Adapt reproducer to include the needed changes to support openshift as workload on RHOSP18.

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

rlobillo commented on scenarios/reproducers/validated-architecture-shiftstack.yml:
```diff
@@ -0,0 +1,159 @@
+      sha256_image_name: "{{ cifmw_discovered_hash }}"
+      image_local_dir: "{{ cifmw_basedir }}/images/"
+      disk_file_name: "base-os.qcow2"
+      disksize: 200
+      memory: 50
+      cpus: 8
```
different from default validated-architecture-1.yml

cjeanner replied:
if CPU is the only difference, we should consider doing the same thing done regarding the compute amount and allow passing it as an inline parameter

rlobillo replied:
it's CPU, memory and disk. I'm creating the variables for these 3. Moved these vars to shift-on-stack-overrides.yml as suggested

rlobillo commented on scenarios/reproducers/validated-architecture-shiftstack.yml:
```diff
@@ -0,0 +1,159 @@
+  cephfs_enabled: true
+
+# OCP on OSP needs to access OSP API from the VMs that behaves as OCP nodes
+cifmw_allow_vms_to_reach_osp_api: true
+# HCI requires bigger size to hold OCP on OSP disks
+cifmw_block_device_size: 100G
```
different from default validated-architecture-1.yml

cjeanner replied:
this may be passed as a parameter, no need to copy the whole file just for that diff.

rlobillo replied:
Moved the var to shift-on-stack-overrides.yml as suggested

rlobillo commented on reproducer.yml:
```diff
@@ -55,3 +55,9 @@
     - role: ci_setup
     - role: reproducer
+
+  post_tasks:
+    - name: Allow traffic from OSP VMs to OSP API (needed for shiftstack)
+      ansible.builtin.shell: iptables -I LIBVIRT_FWI 1 -o ocpbm -j ACCEPT
+      become: true
+      when: cifmw_allow_vms_to_reach_osp_api
```
This is needed for allowing the VMs to contact the OSP API, which is a requirement for the shiftstack OCP nodes. This workload needs to manage OSP resources during both the installation process and the shiftstack operators activity.

cjeanner replied:
I'd rather not use `shell` here:
- there's a [module for iptables](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/iptables_module.html) ( ansible-lint will crash for this)
- if really, `ansible.builtin.command` is enough (ansible-lint will crash for this)

Also, the `when` condition will fail if the parameter isn't set, so you want `| default(false) | bool`.
Maybe it should be done in any cases? Not really sure what this would allow, to be fair.

rlobillo replied:
iptables module does not look to support the out_interface param for LIBVIRT_FWI (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/iptables_module.html#parameter-out_interface). Moving to command module.

cjeanner replied:
dang... do they really filter the chains?

cjeanner commented on scenarios/reproducers/validated-architecture-shiftstack.yml:
```diff
@@ -0,0 +1,159 @@
+  worker_vcpu: 10
+  num_extra_workers: 0
+# Required for egress traffic from pods to the osp_trunk network
+cifmw_devscripts_enable_ocp_nodes_host_routing: true
```
ok, this is a the new param to enable the firewall - no need for a full copy-paste...

rlobillo replied:
Moved the var to shift-on-stack-overrides.yml as suggested

cjeanner reviewed (changes requested):
While I understand the need, I think this PR is wrong, especially on the scenario file.
You should add the parameter to raise the compute CPU as described in the comment, then create one single file, like "shift-on-stack-overrides.yml" and just add the specific overrides there. It would make the whole thing far, far easier to maintain...

rlobillo commented:
> While I understand the need, I think this PR is wrong, especially on the scenario file. You should add the parameter to raise the compute CPU as described in the comment, then create one single file, like "shift-on-stack-overrides.yml" and just add the specific overrides there. It would make the whole thing far, far easier to maintain...

Thanks for your comments, @cjeanner 

I addressed your comments and we are working on validate the changes. Please kindly take another look to the latest.

cjeanner commented on scenarios/reproducers/shift-on-stack-overrides.yml:
```diff
@@ -0,0 +1,8 @@
+cifmw_allow_vms_to_reach_osp_api: true
+# HCI requires bigger size to hold OCP on OSP disks
+cifmw_block_device_size: 100G
+
```
that empty line will crash ansible-lint

cjeanner commented on scenarios/reproducers/shift-on-stack-overrides.yml:
```diff
@@ -0,0 +1,8 @@
+cifmw_libvirt_manager_compute_amount: 3
```
this is the default value for the amount - no need to repeat it in this case

cjeanner commented on reproducer.yml:
```diff
@@ -55,3 +55,10 @@
+
+  post_tasks:
+    - name: Allow traffic from OSP VMs to OSP API (needed for shiftstack)
+      ansible.builtin.command:
```
you'll need a `  # noqa: command-instead-of-module` in order to pass ansible-lint.

cjeanner commented on reproducer.yml:
```diff
@@ -55,3 +55,10 @@
+    - name: Allow traffic from OSP VMs to OSP API (needed for shiftstack)
+      ansible.builtin.command:
+        cmd: iptables -I LIBVIRT_FWI 1 -o ocpbm -j ACCEPT
+      become: true
```
care to add the `become: true` and `when: ...` just under the `name:` key? It makes things slightly more readable when we scroll over a receipt :).

rlobillo replied:
Done

cjeanner commented:
/ok-to-test

cjeanner commented:
/test all

rlobillo commented:
/test all

cjeanner commented on scenarios/reproducers/shift-on-stack-overrides.yml:
```diff
@@ -0,0 +1,6 @@
+cifmw_libvirt_manager_compute_cpus: 8
+cifmw_allow_vms_to_reach_osp_api: true
+# HCI requires bigger size to hold OCP on OSP disks
+cifmw_block_device_size: 100G
```
I think this isn't the right parameter name? Shouldn't it be something like:
```YAML
cifmw_libvirt_manager_compute_disksize: 100
```

rlobillo replied:
Yes, that variable is present in line 1. This is the size that the root disk will have: 200G.

Then, I need to set the variable **cifmw_block_device_size** which is the size for the virtual block device that will used by ceph on the computes: https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/cifmw_block_device/README.md#parameters

So, the outcome will be computes with 200G where 100G will be used by Ceph and the rest will be used by the compute itself.

pablintino commented:
/approve

pablintino commented:
/lgtm
